src/common/buffer.hpp

```diff
@@ -284,104 +284,85 @@ class RxCounters {
     }
 };
 
-class Frame_stats {
+/**
+//  * This class stores the metrics coresponding to a Frame.
+//  * Specifically, it contains a) the number of symbols per frame and b) the number of tasks per symbol, per frame.
+//  */
+class FrameCounters {
 public:
     size_t max_symbol_count;
-    bool last_symbol(int frame_id)
-    {
-        const size_t frame_slot = frame_id % TASK_BUFFER_FRAME_NUM;
-        if (++symbol_count[frame_slot] == max_symbol_count) {
-            symbol_count[frame_slot] = 0;
-            return true;
-        }
-        return false;
-    }
+    size_t max_task_count;
 
-    void init(int _max_symbol_count)
+    void init(
+        int max_symbol_count, int max_task_count = -1, int max_data_symbol = -1)
     {
+        this->max_symbol_count = max_symbol_count;
+        this->max_task_count = max_task_count;
         symbol_count.fill(0);
-        max_symbol_count = _max_symbol_count;
+        //define task counters if #tasks is > 0
```

Comment:
We need to add space between // and define and capitalize the first letter
```suggestion
        // Define task counters if #tasks is > 0
```